packages/framework/src/Model/Payment/SpaydValidator.php

```diff
@@ -0,0 +1,221 @@
+
+class SpaydValidator
+{
+    /**
+     * Základní ověření SPAYD řetězce.
+     * Vyhodí InvalidArgumentException s popisem problému při nevalidním vstupu.
+     *
+     * @param string $spayd
+     */
+    public function validate(string $spayd): void
+    {
+        $spayd = trim($spayd);
+
+        if ($spayd === '') {
+            throw new InvalidArgumentException('SPAYD string is empty.');
+        }
+
+        $parts = explode('*', $spayd);
+
+        if (count($parts) < 3) {
+            throw new InvalidArgumentException('SPAYD string is incomplete.');
+        }
+
+        if ($parts[0] !== 'SPD') {
+            throw new InvalidArgumentException('SPAYD must start with "SPD".');
+        }
+
+        $version = $parts[1];
+
+        if (!in_array($version, ['1.0', '1.1'], true)) {
+            throw new InvalidArgumentException(sprintf('Unsupported SPAYD version "%s".', $version));
+        }
+
+        $fields = $this->parseFields(array_slice($parts, 2));
+
+        // Povinná pole
+        foreach (['ACC', 'AM', 'CC'] as $requiredKey) {
+            if (!array_key_exists($requiredKey, $fields)) {
+                throw new InvalidArgumentException(sprintf('Missing required SPAYD field: %s.', $requiredKey));
+            }
+        }
+
+        // ACC: IBAN [+BIC] – validace IBAN (obecně) a případného BIC
+        $acc = $fields['ACC'];
+        // Pouze IBAN nebo IBAN+BIC oddělené znakem '+'
+        $ibanPart = $acc;
+
+        if (str_contains($acc, '+')) {
+            [$ibanPart, $bic] = explode('+', $acc, 2);
+            $this->assertValidBic($bic);
+        }
+        $ibanPart = strtoupper($ibanPart);
+        $this->assertValidIban($ibanPart);
+
+        // AM: kladná částka, max 2 desetinná místa
+        $amountRaw = $fields['AM'];
+        $amountNormalized = str_replace(',', '.', $amountRaw);
+
+        if (!preg_match('/^\d+(?:\.\d{1,2})?$/', $amountNormalized)) {
+            throw new InvalidArgumentException('Invalid AM (amount) format. Expect digits with optional dot and up to 2 decimals.');
+        }
+        $amount = (float)$amountNormalized;
+
+        if (!is_finite($amount) || $amount <= 0.0) {
+            throw new InvalidArgumentException('AM (amount) must be a positive number.');
+        }
+
+        // CC: 3písmenný ISO kód měny
+        $cc = $fields['CC'];
+
+        if (!preg_match('/^[A-Z]{3}$/', $cc)) {
+            throw new InvalidArgumentException('CC (currency) must be a 3-letter uppercase ISO code.');
+        }
+
+        // X-VS: volitelné, 1–10 číslic dle CZ/SK standardu
+        if (isset($fields['X-VS']) && !preg_match('/^\d{1,10}$/', $fields['X-VS'])) {
+            throw new InvalidArgumentException('X-VS (variable symbol) must be 1 to 10 digits.');
+        }
+    }
+
+    /**
+     * Praktická varianta, která místo výjimek vrací boolean.
+     *
+     * @param string $spayd
+     * @return bool
+     */
+    public function isValid(string $spayd): bool
+    {
```

Comment:
this is used in tests only. I can live with that but if you want, you can remove it, and just use that validate method in tests and assert whether the exception is thrown